taskipy/task_runner.py

```diff
@@ -1,27 +1,46 @@
+import sys
+
+import psutil  # type: ignore
+import signal
 import subprocess
 from pathlib import Path
 from typing import List, Union
 
-from taskipy.task import Task
 from taskipy.pyproject import PyProject
+from taskipy.task import Task
 
 
 class TaskRunner:
     def __init__(self, pyproject_path: Union[str, Path]):
         self.__pyproject_path = pyproject_path
         self.__project = PyProject(pyproject_path)
+        self.__is_macos = sys.platform == 'darwin'
 
     def run(self, task_name: str, args: List[str]) -> int:
         task = Task(task_name, args, self.__project)
 
         for command in task.commands:
             if task.runner is not None:
                 command = f'{task.runner} {command}'
-
             process = subprocess.Popen(
                 command, shell=True, cwd=Path(self.__pyproject_path).parent
             )
 
+            def send_signal_to_task_process(signum: int, frame) -> None:  # pylint: disable=unused-argument
+                # pylint: disable=W0640
+                psutil_process_wrapper = psutil.Process(process.pid)
+
+                if self.__is_macos:
```

Comment:
since this is the only place where we need to check the platform, let's directly do so here, and keep all the relevant contained :).
```suggestion
                if sys.platform == 'darwin':
```
or even further, we can describe what's our intention behind the comparison:
```suggestion
                is_direct_subprocess_a_shell_process = sys.platform != 'darwin'
                if is_direct_subprocess_a_shell_process:
                    # linux code here
                else:
                    # macos
```